### PR TITLE
Fix OpenAPI3 type property should always be set when nullable property is present

### PR DESCRIPTION
**Pull request by skywing918**
https://github.com/microsoft/typespec/issues/4584

### CODE REVIEW

File: packages/openapi3/src/schema-emitter.ts [hunk 1]
```diff
@@ -589,13 +600,14 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
       return wrapWithObjectBuilder(schemaMembers[0], { mergeUnionWideConstraints: true });
     }
 
+    const isMerge = checkMerge(schemaMembers);
     const schema: OpenAPI3Schema = {
       [ofType]: schemaMembers.map((m) =>
-        wrapWithObjectBuilder(m, { mergeUnionWideConstraints: false }),
+        wrapWithObjectBuilder(m, { mergeUnionWideConstraints: isMerge }),
```
Comment (timotheeguerin) [inline packages/openapi3/src/schema-emitter.ts:606]: isn't that going to create unecessary warpping?

can you add a test like below that also check this union `Foo | string | null`

Comment (skywing918) [reply packages/openapi3/src/schema-emitter.ts:606]: I have check this, no more warpping. also added new case
